backend/entityservice/tasks/comparing.py

```diff
@@ -126,34 +122,60 @@ def _create_work_chunks(blocks, dp_block_sizes, dp_ids, log, chunk_size_aim=Conf
         - block_id - The block of encodings this chunk is for
         - range - The range of encodings within the block that make up this chunk.
     """
-    chunks = []
-    for block_id in blocks:
-        log.debug("Chunking block", block_id=block_id)
-        # Note there might be many pairs of datasets in this block
-        for dp1, dp2 in blocks[block_id]:
+    packages = []
+    MAX_CHUNKS_PER_PACKAGE = 320000
+
+    def next_block(blocks):
+        try:
+            block_id, (dp1, dp2) = next(blocks)
             size1 = dp_block_sizes[dp1][block_id]
             size2 = dp_block_sizes[dp2][block_id]
-
-            comparisons = size1 * size2
-
+            num_comparisons = size1 * size2
+            return block_id, dp1, dp2, size1, size2, num_comparisons
+        except StopIteration as e:
+            raise e
+
+    block_name, dp1, dp2, size1, size2, comparisons = next_block(blocks)
+    cur_package = []
+    cur_comparisons = 0
+    try:
+        while True:
```

Comment:
Why not iterate directly over `next_block` (after converting it to a generator)? Then you could avoid using exceptions for control flow